python/ray/serve/tests/test_cli.py

```diff
@@ -731,13 +731,12 @@ def check_application_status():
         status = yaml.safe_load(cli_output)["applications"]
         # TODO(zcin): fix error handling in the application state manager for
         # invalid override options and check for `DEPLOY_FAILED` here.
-        return (
+        assert (
```

Comment:
why assert instead of return OOC?

Reply:
What is OOC? 😅
To answer "why assert" though, I found recently that it's really useful because `wait_for_condition` will print the most recent traceback if an exception occurs. So in the logs it will say something more specific like
```
Last exception: Traceback (most recent call last):
E         File "/Users/cindyz/ray/python/ray/_private/test_utils.py", line 556, in wait_for_condition
E           if condition_predictor(**kwargs):
E         File "/Users/cindyz/ray/python/ray/serve/tests/test_cli.py", line 734, in check_application_status
E           assert (
E       AssertionError: assert ('RUNNING' == 'RUNNING'
E           RUNNING and 'DEPLOYING' == 'DEPLOY_FAILED'
E         - DEPLOY_FAILED
E         + DEPLOYING)
```
Instead of just
```
RuntimeError: The condition wasn't met before the timeout expired.
```

This typically is really helpful for me to debug flaky tests or failures.

Reply:
Oh nice, this is good to know. That looks way more helpful when there are flakes, we should do it across the board.

oh, and it's "out of curiousity." maybe this is some geriatric 90s internet slang

Reply:
can you PSA this on serve-eng slack channel?